earth2studio/models/px/dlesym.py

```diff
@@ -1060,7 +1106,89 @@ def coords_to_ll(self, coords: CoordSystem) -> CoordSystem:
             ll_coords.move_to_end(dim)
         return ll_coords
 
-    # @batch_func()
+    def _nan_interpolate_sst(
+        self, sst: torch.Tensor, coords: CoordSystem
+    ) -> torch.Tensor:
+        """
+        Custom interpolation to fill NaNs over landmasses in SST data.
+        """
+
+        da_sst = xr.DataArray(sst.cpu().numpy(), dims=coords.keys())
+        da_interp = da_sst.interpolate_na(
+            dim="lon", method="linear", use_coordinate=False
+        )
+
+        # Second pass: roll, interpolate along longitude, and unroll
+        roll_amount_lon = int(len(da_interp.lon) / 2)
+        da_double_interp = (
+            da_interp.roll(lon=roll_amount_lon, roll_coords=False)
+            .interpolate_na(dim="lon", method="linear", use_coordinate=False)
+            .roll(lon=len(da_interp.lon) - roll_amount_lon, roll_coords=False)
+        )
+
+        # Third pass do a similar roll along latitude
```

Comment:
Wait roll on lat? Huh? Shouldnt it be reflective padding? Roll would put south poll against the north pole?

Reply:
Wait is this healpix here?

Reply:
I pulled this method directly from the DLESyM project data prep code so haven't thought much about the behavior. It's certainly weird, but this is what the model is trained with. Initializing the forecast with a simple constant-fill over landmasses causes the forecast to eventually diverge

Reply:
The polar behavior seems to work out because both poles are...cold. Goes from this:
![image](https://github.com/user-attachments/assets/18a1fe9e-6d89-4cf0-8b61-f2f44c317fba)
to this:
![image](https://github.com/user-attachments/assets/0f7c187c-ffde-4357-85f9-ad4301782e36)

Reply:
I guess north an south pole are cold, but if this is from the code base good enough for me